uaclient/apt.py

```diff
@@ -48,18 +48,18 @@ def assert_valid_apt_credentials(repo_url, username, password):
     if not os.path.exists("/usr/lib/apt/apt-helper"):
         return
     try:
-        with tempfile.NamedTemporaryFile() as tmpf:
-            util.subp(
-                [
-                    "/usr/lib/apt/apt-helper",
-                    "download-file",
-                    "{}://{}:{}@{}/ubuntu/pool/".format(
-                        protocol, username, password, repo_path
-                    ),
-                    tmpf.name,
-                ],
-                timeout=APT_HELPER_TIMEOUT,
-            )
+        tmpf = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
From [`NamedTemporaryFile`'s docs](https://docs.python.org/3/library/tempfile.html): 

> If delete is true (the default), the file is deleted as soon as it is closed.

So using it as a context manager should result in it being cleaned up regardless of exceptions. I've just tested this:

```py
with NamedTemporaryFile() as f:
    print(f.name)
    input('')
    raise Exception
```
and confirmed that the printed file did exist when the input prompt happened, but did not exist after.

Could we drop this change from this PR, and you can re-propose it separately if I'm misunderstanding what you were seeing was wrong? (Ideally, we would have distinct changes in different PRs regardless.)

Reply:
Added separate issue #755 and withdrew funky NamedTempFile workaround